src/App.tsx

```diff
@@ -1,8 +1,53 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './App.css';
+import { ListOfGoods } from './Components/ListOfGoods/ListOfGoods';
+import { Buttons } from './Components/Buttons/Buttons';
+import { getGoods } from './api/utils/getGoods';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface State {
+  goods: Good[];
+}
 
-export default App;
+export class App extends Component<{}, State> {
+  state = {
+    goods: [],
+  };
+
+  loadAllGoods = () => {
+    getGoods().then(goods => this.setState({ goods }));
+  };
+
+  loadFiveFirstGoods = () => {
+    getGoods().then((goods) => {
+      this.setState(({
+        goods: goods
+          .sort((a, b) => a.name.localeCompare(b.name))
+          .filter(good => good.id < 6),
+      }));
+    });
+  };
+
+  loadRedGood = () => {
+    getGoods().then(goods => {
+      this.setState({
+        goods: goods.filter((good: {color: string}) => good.color === 'red'),
```

Comment:
do not require to describe color type here, also you forgot spaces
```suggestion
        goods: goods.filter((good: { color: string }) => good.color === 'red'),
```